otlp/trace.go

```diff
@@ -257,22 +259,42 @@ func shouldTrimTraceId(traceID []byte) bool {
 	return true
 }
 
-// getSpanStatusCode checks the value of both the deprecated code and code fields
-// on the span status and using the rules specified in the backward compatibility
-// notes in the protobuf definitions. See:
+// evaluateSpanStatus returns the integer value of the span's status code and
+// a bool for whether to consider the status an error.
+//
+// The type conversion from proto enum value to an integer is done here because
+// the events we produce from OTLP spans have no knowledge of or interest in
+// the OTLP types generated from enums in the proto definitions.
 //
-// https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L230
-func getSpanStatusCode(status *trace.Status) trace.Status_StatusCode {
+// Until we don't have to process OTLP traces from proto <= v0.11.0, this
+// function checks the value of both the status.Code and status.DeprecatedCode
+// fields on the span according to the rules for backward compatibility.
+//
+// See: https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L230
+func evaluateSpanStatus(status *trace.Status) (int, bool) {
 	if status == nil {
-		return trace.Status_STATUS_CODE_UNSET
+		return int(trace.Status_STATUS_CODE_UNSET), false
 	}
-	if status.Code == trace.Status_STATUS_CODE_UNSET {
+
+	isError := false
+	status_code := int(status.Code)
+
+	switch status.Code {
+	case trace.Status_STATUS_CODE_OK:
+		// OK, thumbs up. Let's do this.
+	case trace.Status_STATUS_CODE_ERROR:
+		isError = true
+	case trace.Status_STATUS_CODE_UNSET:
+		//lint:ignore SA1019 keep DepCode until we move to proto v0.12.0+
```

Comment:
I don't think we need the first part of this if statement, status_code will already be `0` from line 280.

Reply:
🤔 Yup. I agree.

Only need the `status.DeprecatedCode != trace.Status_DEPRECATED_STATUS_CODE_OK` else case